pkgs/applications/misc/wspr/default.nix

```diff
@@ -0,0 +1,57 @@
+{ stdenv, fetchFromGitHub, pkgconfig, gfortran, portaudio
+, pythonFull, swig2, pythonPackages, fftwFloat, libsamplerate, tk
+
+}:
+
+stdenv.mkDerivation rec {
+  name = "wspr-${version}";
+  version = "4.0";
+
+  buildInputs = [ pkgconfig gfortran portaudio fftwFloat libsamplerate tk pythonFull (with pythonPackages; [ numpy tkinter pmw pillow ])];
+
+#  propagatedBuildInputs = [
+#    numpy scipy matplotlib
+#  ];
+
+  propagatedBuildInputs = [
+    pythonPackages.tkinter
+    pythonPackages.pmw
+    pythonPackages.numpy
+    pythonPackages.pillow
+  ];
+
+  src = fetchFromGitHub {
+    owner = "jj1bdx";
+    repo = "wspr";
+    rev = "78cf9932e75446e0a8face90570674cf5ddad9f4";
+    sha256 = "0fxip023cgq3223ahr17fnmxqmjb9iy994xjbfvf69g5jwqjnprp";
+  };
+
+  configureFlags = [ "--with-portaudio-include-dir=${portaudio}/include --with-portaudio-lib-dir=${portaudio}/lib" ];
+ 
+  installPhase = ''
+    rm -rf  build
+    grep -rl Numeric . | xargs sed -i 's/Numeric/numpy/'
```

Comment:
The best way is to generate a git patch for all these ad-hoc changes here and then include apply it via `patches = [ ./my-patch.patch ];`

You can generate a git patch by checking out the repository, doing all necessary changes, committing them, then running `git format-patch @^`. This will export the latest commit as a patch file usable in nix.